### ISSUE TITLE
Shift checkpoint files from test_chkpt to tmp

### ISSUE THREAD

**Issue by Chiffafox** [x94carbone/GGCE#64]
Right now some of the tests -- specifically, test_chkpt -- are saving some files to disk in the head directory, then deleting them with shutil.rmtree. The proper way of handling this is is with a tmp_path.

**Comment by x94carbone** [comment]
@Chiffafox yep there's even a [standard library class](https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory) for this. It's really great when used in a context manager.
